be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -317,6 +345,7 @@ Status ScanOperator::_pickup_morsel(RuntimeState* state, int chunk_source_index)
             _is_finished = true;
             return status;
         }
+        need_detach = false;
```

Comment:
Why need_detach is false here?

Reply:
In normal case, we need only attach a chunk_source; but in exception(the chunk_source does not create successfully), we need to detach here.

Attatch/Detech implements a simple ref-counting mechanism, which is used to help the `ScanOperator` to check is there any running `ChunkSource`.